applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -454,6 +601,35 @@ class GeometryUtilities
         }
     }
 
+    // --------------------------------------------------------------------------
+    SizeType GetNumberOfElementalNodesInModelPart(const ModelPart& rModelPart) const
```

Comment:
I can fool this algorithm by giving you quads and tetras ;)

I recommend checking the geometry type instead

Reply:
True.... did not think about it. I'll update :)